apps/prairielearn/assets/scripts/lib/questionsTable.ts

```diff
@@ -0,0 +1,367 @@
+import { decodeData, onDocumentReady, parseHTMLElement } from '@prairielearn/browser-utils';
+import {
+  Tabulator,
+  FormatModule,
+  EditModule,
+  FilterModule,
+  SortModule,
+  PageModule,
+  MutatorModule,
+  FrozenColumnsModule,
+  CellComponent,
+  Editor,
+} from 'tabulator-tables';
+import { html } from '@prairielearn/html';
+import { uniq } from 'lodash';
+
+import type {
+  Topic,
+  Tag,
+  SharingSet,
+  AssessmentsFormatForQuestion,
+} from '../../../src/lib/db-types';
+import type { QuestionsPageDataAnsified } from '../../../src/models/questions';
+import { idsEqual } from '../../../src/lib/id';
+
+// Copied from QuestionTable.html.ts component. Not importing because the import process creates dependencies of files that are not compatible with tsc.
+interface EncodedQuestionsData {
+  plainUrlPrefix: string;
+  urlPrefix: string;
+  questions: QuestionsPageDataAnsified[];
+  course_instances: {
+    id: string;
+    short_name: string | null;
+    current: boolean;
+  }[];
+  showSharingSets: boolean;
+  qidPrefix: string | undefined;
+}
+
+Tabulator.registerModule([
+  FormatModule,
+  EditModule,
+  FilterModule,
+  SortModule,
+  PageModule,
+  MutatorModule,
+  FrozenColumnsModule,
+]);
+
+onDocumentReady(() => {
+  const { plainUrlPrefix, questions, course_instances, showSharingSets } =
+    decodeData<EncodedQuestionsData>('questions-table-data');
+  const table = new Tabulator('#questionsTable', {
+    data: questions,
+    layout: 'fitData',
+    pagination: true,
+    paginationCounter: (pageSize, currentRow, _currentPage, totalRows) =>
+      `Showing ${currentRow}-${currentRow + pageSize - 1} of ${totalRows} question${
+        totalRows === 1 ? '' : 's'
+      }` +
+      (totalRows === questions.length
+        ? ''
+        : ` (filtered from ${questions.length} total question${
+            questions.length === 1 ? '' : 's'
+          })`),
+    paginationSize: 50,
+    paginationSizeSelector: [10, 20, 50, 100, 200, 500, true],
```

Comment:
For a future PR: some of these options, as well as a few settings below, could be moved to a separate library file that covers the default settings for pages with tabulator. This library could also have some basic custom formatters/filters, as well as handlers for Ctrl-F, column visibility and width rearrangement.

Reply:
It seems like you did this in 3df3e45?

Reply:
Indeed. I did not create the custom formatters/filters, but the rest is there. I'm still considering if I should create formatters/filters for e.g., badge lists, by adding [custom reusable items](https://tabulator.info/docs/6.2/modules#module-format) that are added to the built-in ones.